### PR TITLE
Skill Recommendation

### PR DESCRIPTION
**Pull request by AndreasVarotsis**
Adds four endpoints for skill recommendation functions.

- create_skill_similarity_matrix - creates comparison matrix for similar skills
- create_job_embedding_matrix- creates comparison matrix for skills based on title similarity

Once these are run, you can then run 
- me_combined_skill_recommend that will provide a mixed list of 10 recommended skills 

or to get components individually:
- skill_recommender - given a skill, recommends other similar skills
- me_job_title_recommender - given a job title, recommends similar skills

### CODE REVIEW

File: api/ellandi/registration/recommend.py [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+
+
+def get_similar_embedding_jobs(job_title, job_embeddings):
+    """given a series of job embeddings, a new title and optionally top N, returns most similar jobs"""
+
+    skill_return_count = 10
+    model_name = "all-MiniLM-L6-v2"
+
+    model = SentenceTransformer(model_name)
+    job_embedding = model.encode(job_title).reshape((1, 384))
```
Comment (nmenezes0) [inline api/ellandi/registration/recommend.py:65]: Where is the 384 coming from? Will it always be 384? (Same thing in one of functions below too.)

File: api/ellandi/registration/recommend.py [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+    model = SentenceTransformer(model_name)
+    # Sentences are encoded by calling model.encode()
+    embeddings = model.encode(job_titles)
+    embedding_df = pd.DataFrame(index=job_titles, data=embeddings)
+
+    return embedding_df
+
+
+def get_similar_embedding_jobs(job_title, job_embeddings):
+    """given a series of job embeddings, a new title and optionally top N, returns most similar jobs"""
```
Comment (nmenezes0) [inline api/ellandi/registration/recommend.py:59]: Is this docstring right - don't see an optional top N?

Comment (nmenezes0) [reply api/ellandi/registration/recommend.py:59]: Also - is this function actually used anywhere?

File: api/ellandi/registration/views.py [hunk 3]
```diff
@@ -497,3 +504,64 @@ def me_suggested_skills(request):
...
+    return Response(status=status.HTTP_200_OK)
+
+
+@extend_schema(methods=["GET"], request=serializers.SkillTitleSerializer())
+@decorators.api_view(["GET"])
+@decorators.permission_classes(
+    (permissions.AllowAny,)
+)  # TODO - I think this is fine for permissions - anyone can see recommendation?
+def skill_recommender(request, skill_name):
+    # requires create_skill_similarity_matrix endpoint to have been run first
```
Comment (nmenezes0) [inline api/ellandi/registration/views.py:537]: Assume you will handle the case where the matrix/file doesn't exist (can look at some of the other endpoints to see how to return a response that's not 200/OK).

File: api/ellandi/registration/recommend.py (hunk 2 above)
Comment (AndreasVarotsis) [reply api/ellandi/registration/recommend.py:59]: Good point, killed off

File: api/ellandi/registration/recommend.py (hunk 1 above)
Comment (AndreasVarotsis) [reply api/ellandi/registration/recommend.py:65]: Number of vectors...have cleaned up!

**Comment by Singletoned** [comment]
Don't merge `develop` in.  Rebase it on `develop`

**Comment by Singletoned** [comment]
```
git pull origin develop
git push origin feature/recommend_skills --force-with-lease
```

**Review by Singletoned — APPROVED** [review]

**Pull request merged**